vars/buildPlugin.groovy

```diff
@@ -61,7 +61,14 @@ def call(Map params = [:]) {
     }
 
     tasks[stageIdentifier] = {
-      retry(count: 3, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
+      int retryCounts = 1
+      retry(count: 2, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
```

Comment:
You need to add the `agent()` condition to map what we have in ATH: https://github.com/jenkinsci/acceptance-test-harness/blob/ecaf4fc38523b77e0997daa4285999965276a3e2/Jenkinsfile#L116

I don't know if it is cumulative or mutually exclusive with kubernetesAgent though: if we can only spcify on of each, we'll need a tiny function with a closure to sleect retrys based on a condition, and then run the body.

Reply:
Or maybe not needed thanks to `handleNonKubernetes` (ref. https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#retry-retry-the-body-up-to-n-times) 🤔

Reply:
I think that it's why `handleNonKubernetes` is for :  https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#retry-retry-the-body-up-to-n-times
```
    handleNonKubernetes : boolean (optional)
    Behave like the generic agent() (Agent errors) when applied to a non-Kubernetes agent. Useful in cases where it is hard to predict in a job definition whether a Kubernetes or other sort of agent will be used.
```

Reply:
o sorry, the comments were not uptodate